### PR TITLE
feat: widget and embed styles prop (NV-1385)

### PR DESCRIPTION
**Pull request by LetItRock**
### What change does this PR introduce? 

Added the option to pass the new `styles` prop for the iframe embed.

Example usage:
```html
<script>
      (function (n, o, t, i, f) {
        (n[i] = {}), (m = ['init']);
        n[i]._c = [];
        m.forEach(
          (me) =>
            (n[i][me] = function () {
              n[i]._c.push([me, arguments]);
            })
        );
        var elt = o.createElement(f);
        elt.type = 'text/javascript';
        elt.async = true;
        elt.src = t;
        var before = o.getElementsByTagName(f)[0];
        before.parentNode.insertBefore(elt, before);
      })(window, document, 'http://.../embed.umd.min.js', 'novu', 'script');

      novu.init(
        'APP_ID',
        {
          unseenBadgeSelector: '#unseen-badge',
          bellSelector: '#notification-bell',
          styles: {
            header: {
              root: {
                backgroundColor: 'green',
                '&:hover': { backgroundColor: 'pink' },
              },
            },
          },
        },
        {
          subscriberId: 'SUBSCRIBER_ID',
        }
      );
    </script>
```

### Other information (Screenshots)

https://user-images.githubusercontent.com/2607232/208481932-94572a73-f2f2-48ba-a8f0-ed3f6e2c3528.mov

### CODE REVIEW

File: docs/docs/notification-center/iframe-embed.md [hunk 1]
```diff
@@ -174,12 +174,13 @@ novu.init(
...
+| `position.top`        | `string` \| `number` | Override the top position of the notification center drop down                                                                                                                                                        |
+| `position.left`       | `string` \| `number` | Override the left position of the notification center drop down                                                                                                                                                       |
+| `theme`               | `object`             | Provide a custom theme for the notification center to use (for example see [above](#customizing-the-theme))                                                                                                           |
+| `styles`              | `object`             | Provide a custom styles for the notification center. Please note that the limitation is that you can't use the style function as a prop to override the component styles. Read more about [styling](#custom-styling). |
```
Comment (LetItRock) [inline docs/docs/notification-center/iframe-embed.md:186]: updated the docs about passing the `styles` prop in the `init` function argument

File: apps/widget/src/components/notification-center/NotificationCenterWidget.tsx [hunk 2]
```diff
@@ -2,7 +2,7 @@ import { useEffect, useState } from 'react';
 import * as WebFont from 'webfontloader';
 import { css, Global } from '@emotion/react';
 import { NotificationCenter, NovuProvider, ITranslationEntry, ITab, IStore } from '@novu/notification-center';
-import { INovuThemeProvider } from '@novu/notification-center';
+import type { INovuThemeProvider, INotificationCenterStyles } from '@novu/notification-center';
```
Comment (LetItRock) [inline apps/widget/src/components/notification-center/NotificationCenterWidget.tsx:5]: renamed the interface to `INotificationCenterStyles`

File: libs/embed/src/embed.ts [hunk 3]
```diff
@@ -20,6 +20,8 @@ class Novu {
 
   private theme?: Record<string, unknown>;
 
+  private styles?: INotificationCenterStyles;
```
Comment (LetItRock) [inline libs/embed/src/embed.ts:23]: the option on the `embed` script

**Review by scopsy — APPROVED** [review]
Good bless you

**Pull request merged**